scripts/prove_rpc.sh

```diff
@@ -56,41 +55,27 @@ RUN_VERIFICATION="${RUN_VERIFICATION:-false}"
 # Recommended soft file handle limit. Will warn if it is set lower.
 RECOMMENDED_FILE_HANDLE_LIMIT=8192
 
-mkdir -p $PROOF_OUTPUT_DIR
+mkdir -p "$PROOF_OUTPUT_DIR"
 
-if $IGNORE_PREVIOUS_PROOFS ; then
-    # Set checkpoint height to previous block number for the first block in range
-    prev_proof_num=$(($1-1))
-    PREV_PROOF_EXTRA_ARG="--checkpoint-block-number ${prev_proof_num}"
+if [ -n "$CHECKPOINT_BLOCK" ] ; then
+    # Checkpoint block provided, pass it to the prover as a flag
+    PREV_PROOF_EXTRA_ARG="--checkpoint-block $CHECKPOINT_BLOCK"
 else
+    # Checkpoint block not provided, but is required hash-based start block
+    if [[ $START_BLOCK == 0x* ]]; then
+        echo "Checkpoint block is required when specifying blocks by hash"
+        exit 1
+    fi
+
+    # Checkpoint block not provided, deduce proof starting point from the start block
     if [[ $1 -gt 1 ]]; then
         prev_proof_num=$(($1-1))
         PREV_PROOF_EXTRA_ARG="-f ${PROOF_OUTPUT_DIR}/b${prev_proof_num}.zkproof"
     fi
 fi
 
-# Convert hex to decimal parameters
-if [[ $START_BLOCK == 0x* ]]; then
```

Comment:
Some check if $START_BLOCK/$END_BLOCK is actually of block hash length if it starts with `0x`  and early exit on error would be nice.

Reply:
Have added the check now 👍 

In general I would suggest that this script be replaced with a rust program (or if you like to "script" with Go or JS/TS etc). If the script won't continue to grow, it would be not worth the effort. But if you envision more use cases and complexity being added here to the benefit of devex / automation, I highly recommend moving from shell to another language (for maintainability, testability, etc).

Reply:
Agreed. In general we use this script for testing, all the checks must be in the Rust input processing anyway.